src/main/java/york/pharmacy/medicines/Medicine.java

```diff
@@ -0,0 +1,42 @@
+package york.pharmacy.medicines;
+
+import jakarta.persistence.*;
+import jakarta.validation.constraints.NotNull;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.springframework.data.annotation.CreatedDate;
+import org.springframework.data.annotation.LastModifiedDate;
+import org.springframework.data.jpa.domain.support.AuditingEntityListener;
+
+import java.time.Instant;
+
+@Entity
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@EntityListeners(AuditingEntityListener.class)
+public class Medicine {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @NotNull
+    private String name;
+
+    @NotNull
+    private String code;
```

Comment:
Is there a way to mark code as unique so the database won't store duplicates?

Reply:
Great point, that is important. It looks like I can implement this annotation to solve this.
```Java
@UniqueField(
        fieldName = "code",
        entityClass = Medicine.class,
        message = "Code must be unique"
 )
 ```